--- a/src/ircam.c
+++ b/src/ircam.c
@@ -147,11 +147,11 @@
 
 	psf->endian = SF_ENDIAN_LITTLE ;
 
-	if (psf->sf.channels > SF_MAX_CHANNELS)
+	if (psf->sf.channels < 1 || psf->sf.channels > SF_MAX_CHANNELS)
 	{	psf_binheader_readf (psf, "Epmf44", 0, &marker, &samplerate, &(psf->sf.channels), &encoding) ;
 
 		/* Sanity checking for endian-ness detection. */
-		if (psf->sf.channels > SF_MAX_CHANNELS)
+		if (psf->sf.channels < 1 || psf->sf.channels > SF_MAX_CHANNELS)
 		{	psf_log_printf (psf, "marker: 0x%X\n", marker) ;
 			return SFE_IRCAM_BAD_CHANNELS ;
 			} ;
@@ -161,7 +161,7 @@
 
 	psf_log_printf (psf, "marker: 0x%X\n", marker) ;
 
-	psf->sf.samplerate = (int) samplerate ;
+	psf->sf.samplerate = psf_lrintf (samplerate) ;
 
 	psf_log_printf (psf,	"  Sample Rate : %d\n"
 							"  Channels    : %d\n"
